src/elexclarity/formatters/results.py

```diff
@@ -0,0 +1,168 @@
+from collections import defaultdict
+from slugify import slugify
+import xmltodict
+
+from elexclarity.formatters.base import ClarityConverter
+from elexclarity.utils import get_list
+
+
+class ClarityDetailXMLConverter(ClarityConverter):
+    """
+    A class to convert Clarity XML into our expected data format.
+    """
+
+    def aggregate_choice_vote_types(self, choice, level):
+        """
+        Takes a Clarity `Choice` object and aggregates all the different
+        kinds of votes into one total per subunit.
+
+        For example, a precinct level file will have an entry like this:
+
+        ```
+        <Choice key="1" text="Donald J. Trump (I) (Rep)" party="NP" totalVotes="4018">
+            <VoteType name="Election Day Votes" votes="431">
+                <Precinct name="Douglas" votes="431" />
+            </VoteType>
+            <VoteType name="Advanced Voting Votes" votes="3099">
+                <Precinct name="Douglas" votes="3099" />
+            </VoteType>
+            <VoteType name="Absentee by Mail Votes" votes="487">
+                <Precinct name="Douglas" votes="487" />
+            </VoteType>
+            <VoteType name="Provisional Votes" votes="1">
+                <Precinct name="Douglas" votes="1" />
+            </VoteType>
+        </Choice>
+        ```
+
+        And a county level file will have an entry like this:
+
+        ```
+        <Choice key="2" text="Joseph R. Biden (Dem)" party="NP" totalVotes="2474507">
+            <VoteType name="Election Day Votes" votes="367205">
+                <County name="Appling" votes="334" />
+                <County name="Atkinson" votes="250" />
+                <County name="Bacon" votes="140" />
+                <County name="Baker" votes="149" />
+                <County name="Baldwin" votes="1527" />
+                <County name="Banks" votes="150" />
+                <County name="Barrow" votes="1717" />
+                <County name="Bartow" votes="2175" />
+                <County name="Ben Hill" votes="336" />
+                ...
+            </VoteType>
+            ...
+        </Choice>
+        ```
+
+        So we have to convert our level into the XML tag name and then aggregate
+        the vote types accordingly.
+        """
+        subunits = defaultdict(lambda: 0)
+        clarity_level = level.capitalize()
+
+        for vote_type in get_list(choice.get("VoteType", [])):
+            vote_type_subunits = vote_type.get(clarity_level, [])
+            for vote_type_subunit in get_list(vote_type_subunits):
+                if level == "precinct":
+                    subunit_id = self.get_precinct_id(vote_type_subunit["name"])
```

Comment:
Do we no longer want the precinct ids to have the county appended to the front?

Reply:
I don't think so. I was originally thinking we'd have a statewide results object with precinct subunits (in which case we'd want county + precinct IDs), but if the county results are the top-level import, then the data importer is going to automatically take that top level ID and use it with the subunit IDs to make something like <date>_GA_G_S_<county fips>_<precinct slug> (thus my making sure we've got the county FIPS in the top level ID)